lib/rpc-client.js

```diff
@@ -0,0 +1,60 @@
+import {
+  FilecoinRpcError,
+  INVALID_RPC_RESPONSE,
+  NO_RESULT_IN_RESPONSE,
+  RPC_ERROR,
+} from './errors.js'
+
+/**
+ * @param {string} method
+ * @param {unknown[]} params
+ * @param {string} rpcUrl
+ * @param {string} rpcAuth
+ * @param {object} [options]
+ * @param {globalThis.fetch} [options.fetch]
+ */
+export async function rpc(method, params, rpcUrl, rpcAuth, { fetch = globalThis.fetch } = {}) {
```

Comment:
Let's move the rpc configuration to the options object. There are too many positional arguments in your proposal.

```suggestion
export async function rpc(method, params, { rpcUrl, rpcAuth, fetch = globalThis.fetch } = {}) {
```

Alternatively, move all parameters to the object.

```suggestion
export async function rpc({ method, params, rpcUrl, rpcAuth, fetch = globalThis.fetch } = {}) {
```

Personally, I prefer the first version `(method, params, options)` because the `options` object usually stays the same for all requests. Such a design is also similar to `fetch(url, options)`.